orderbook/src/trade_events.rs

```diff
@@ -0,0 +1,147 @@
+        //    wrong. In theory this could still happen if the last MAX_REORG_BLOCK_COUNT blocks had
+        //    more than INSERT_TRADE_BATCH_SIZE trade events but this is unlikely.
+        // There alternative solutions for 2. but this one is the most practical. For example, we
+        // could keep all reorgable events in this struct and only store ones that are older than
+        // MAX_REORG_BLOCK_COUNT in the database but then any code using trade events would have to
+        // go through this class instead of being able to work with the database directly.
+        // Or we could make the batch size unlimited but this runs into problems when we have not
+        // updated it in a long time resulting in many missing events which we would all have to
+        // in one transaction.
+        let mut have_deleted_old_events = false;
+        while let Some(trades) = events.next().await {
+            let trades = trades.context("failed to get event")?;
+            if !have_deleted_old_events {
+                self.db
+                    .replace_trades(*range.start(), &[])
+                    .await
+                    .context("failed to replace trades")?;
+                have_deleted_old_events = true;
+            } else {
+                self.db
+                    .insert_trades(trades.as_slice())
+                    .await
+                    .context("failed to insert trades")?;
+            }
+        }
+        Ok(())
+    }
+
+    async fn event_block_range(&self) -> Result<RangeInclusive<u64>> {
+        let web3 = self.web3();
+        let last_handled_block = self
+            .db
+            .block_number_of_most_recent_trade()
+            .await
+            .context("failed to get last handle block")?;
+        let current_block = web3
+            .eth()
+            .block_number()
+            .await
+            .context("failed to get current block")?
+            .as_u64();
+        let from_block = last_handled_block.saturating_sub(MAX_REORG_BLOCK_COUNT);
+        anyhow::ensure!(
+            from_block <= current_block,
+            format!(
+                "current block number according to node is {} which is more than {} blocks in the \
+                 past compared to last handled block {}",
+                current_block, MAX_REORG_BLOCK_COUNT, last_handled_block
+            )
+        );
+        Ok(from_block..=current_block)
+    }
+
+    fn web3(&self) -> Web3<impl Transport> {
+        self.contract.raw_instance().web3()
+    }
+
+    async fn past_events(
+        &self,
+        block_range: &RangeInclusive<u64>,
+    ) -> Result<impl Stream<Item = Result<DbTrade>>, ExecutionError> {
+        Ok(self
+            .contract
+            .all_events()
+            .from_block((*block_range.start()).into())
+            .to_block((*block_range.end()).into())
+            .block_page_size(100)
```

Comment:
Nit: you can use a constant here and I would also choose the value bigger.

Reply:
I didn't feel that extracting this out to a constant helped document it. I will increase it to 1000.